include/souper/Infer/AbstractInterpreter.h

```diff
@@ -39,6 +45,46 @@ namespace souper {
     llvm::KnownBits ule(const llvm::KnownBits &lhs, const llvm::KnownBits &rhs);
     llvm::KnownBits sle(const llvm::KnownBits &lhs, const llvm::KnownBits &rhs);
   }
+
+  bool isConcrete(souper::Inst *I,
+                  bool ConsiderConsts = true,
+                  bool ConsiderHoles = true);
+
+  class KnownBitsAnalysis {
+    std::unordered_map<Inst*, llvm::KnownBits> KBCache;
+
+    // Checks the cache or instruction metadata for knonwbits information
+    bool cacheHasValue(Inst* I);
```

Comment:
can you go through this patch and systematically write "Inst *I" instead of "Inst* I"?

Reply:
should be done now.